stakingModel/contracts/AbstractGoodStaking.sol

```diff
@@ -8,15 +8,35 @@ pragma solidity 0.5.4;
  */
 contract AbstractGoodStaking {
 
+    /**
+     * @dev Structure to store staking details.
+     * It contains amount of tokens staked and blocknumber at which last staked.
+     */
     struct Staker {
         uint256 stakedToken;
         uint256 lastStake;
     }
 
+    /**
+     * @dev Mapping to store staking details for each user.
+     */
     mapping(address => Staker) public stakers;
 
+    /**
+     * @dev Emitted when `staker` stake `value` tokens
+     */
     event Staked(address indexed staker, uint256 value);
+
+    /**
+     * @dev Emitted when `staker` withdraws their stake `value` tokens from 
+     * `actual` contract balance.
+     */
     event StakeWithdraw(address indexed staker, uint256 value, uint256 actual);
```

Comment:
```suggestion
    event StakeWithdraw(address indexed staker, uint256 value, uint256 remainingBalance);
```

Reply:
`actual` represents the contract balance before transferring `value`  to staker. Earlier, it was represented with `daiActual` name. @sirpy

Reply:
still we need to rename it. call it startingBalance

Reply:
ok, will update in next commit

Reply:
or else, if you want we can emit remaining balance in function. let me know what should it be startingBalance or remainingBalance. According to remainingBalance will make much more sense. @sirpy

Reply:
ok remainingBalance

Reply:
ok, will update in next commit.